Allow to connect with gocloud to AWS/GCP instances. (#29)
cyrilgdn/terraform-provider-postgresql: website/docs/index.html.markdown

  Version](https://www.postgresql.org/support/versioning/) or `current`.  Once a
  connection has been established, Terraform will fingerprint the actual
  version.  Default: `9.0.0`.

## GoCloud

> By default, the provider uses [lib/pq][libpq] library to connect to PostgreSQL servers, but you can switch to [GoCloud](https://gocloud.dev/howto/sql/).
> GoCloud simplifies the connection to AWS/GCP hosted databases. For example GCP requires a custom proxy with authentication credentials, GoCloud manages it automatically.

Reviewer: ```suggestion
By default, the provider uses the [lib/pq][libpq] library to directly connect to PostgreSQL host instance. For connections to AWS/GCP hosted instances, the provider can connect through the [GoCloud](https://gocloud.dev/howto/sql/) library. GoCloud simplifies connecting to AWS/GCP hosted databases, managing any proxy or custom authentication details.
```

### AWS

> To enable GoCloud for AWS RDS, set `scheme` to `awspostgres`, `host` is the database endpoint in RDS.

Reviewer: ```suggestion
To enable GoCloud based connections to AWS RDS instances, set `scheme` to `awspostgres` and `host` to the RDS database's endpoint value.
```

(e.g.: `instance.xxxxxx.region.rds.amazonaws.com`)

> 

Reviewer: Although I haven't tried it myself, similarly to Google I presume you need to setup credentials for RDS connections. My understanding is that like GCP, it uses a Platform (GCP/AWS) with the right permissions to create that connection tunnel.

Author: AWS does not have a mandatory proxy like GCP, so with public accessible instance this example work. Of course most user will probably need to use a tunnel but I'm not sure if Gocloud could help for that.

```hcl
provider "postgresql" {
  scheme   = "awspostgres"
  host     = "test-instance.cvvrsv6scpgd.eu-central-1.rds.amazonaws.com"
  username = "postgres"
  port     = 5432
  password = "test1234"

  superuser = false
}
```

### GCP

> To enable GoCloud for GCP SQL, set `scheme` to `gcppostgres`, host is the connection name in following format: `project/region/instance`

Reviewer: ```suggestion
To enable GoCloud for GCP SQL, set `scheme` to `gcppostgres` and `host` to the connection name of the instance in following format: `project/region/instance`.
```
